sdk/core/azure-core/src/datetime.cpp

```diff
@@ -777,6 +785,77 @@ DateTime DateTime::FromString(std::string const& dateString, DateFormat format)
     throw DateTimeException("Unrecognized date format.");
   }
 
-  result.m_interval = static_cast<IntervalType>(secondsSince1601 * TicksPerSecond + fracSec);
+  result.m_since1601.m_100nsIntervals
+      = static_cast<int64_t>(secondsSince1601 * Duration::IntervalsOf100nsPerSecond + fracSec);
   return result;
 }
+
+DateTime::DateTime(int year, int month, int day, int hour, int minute, int seconds)
+{
+  // We should combine creation/validation logic, so it is reusable from both parsing functions and
+  // from this constructor
+  if (year > MaxYear || year < MinYear)
+  {
+    throw DateTimeException(
+        year > MaxYear ? "The requested year exceeds the year 9999."
+                       : "The requested year is less than the year 1601.");
+  }
+
+  if (month <= 0 || month > 12)
+  {
+    throw DateTimeException("Invalid month number.");
```

Comment:
nit: consistency with other messages below
```suggestion
    throw DateTimeException("Invalid month value.");
```